src/functions/forecast/list-projects/handler.ts

```diff
@@ -47,22 +48,74 @@ const listProjects = async (api: ForecastApiService, currentDate: Date, paramete
       startDate: project.start_date,
       endDate: project.end_date,
       status: project.status,
-      stage: project.stage
+      stage: project.stage,
+      color: project.color
     }
   })
 } 
 
+/**
+ * Gets project by id
+ * 
+ * @param api Instance of ForecastApiService
+ * @param parameters Parameters
+ * @returns A single project
+ */
+const listProject = async (api: ForecastApiService, parameters: ListProjectsParameters): Promise<Response> => {
+  const project = await api.getProject(parameters.projectId);
+  
+  return {
+    id: project.id,
+    name: project.name,
+    startDate: project.start_date,
+    endDate: project.end_date,
+    status: project.status,
+    stage: project.stage,
+    color: project.color
+  }
+}
+
 /**
  * Lambda for listing Forecast projects
  * 
  * @param event event
  */
 const listProjectsHandler: ValidatedEventAPIGatewayProxyEvent<any> = async event => {
   const api = CreateForecastApiService();
+  let startDateParameter = new Date();
+  let endDateParameter = new Date();
 
+  if (event.queryStringParameters.projectId){
+    const project = await listProject(api, {projectId: event.queryStringParameters.projectId});
+    if (!project.id) {
+      return {
+        statusCode: 400,
```

Comment:
I think 204 might be a more appropriate code in this case. Have a look and see what you think